llvm/lib/Target/DirectX/DXILOpLowering.cpp

```diff
@@ -770,8 +770,14 @@ class OpLowerer {
         continue;
       Intrinsic::ID ID = F.getIntrinsicID();
       switch (ID) {
-      default:
+      case Intrinsic::dx_resource_casthandle:
+      case Intrinsic::not_intrinsic:
         continue;
+      default: {
+        DiagnosticInfoUnsupported Diag(F, "Unknown intrinsic?");
```

Comment:
We should probably include the name of the intrinsic in the error message here. It'd also be nice to have a debug location, but I suspect there isn't really anywhere we can grab a reasonable one from so I guess we'll do without.

Reply:
The intrinsic name seems to already be encoded this is the error string i see `function llvm.vector.reduce.and.v4i32 i32 (<4 x i32>): Unknown intrinsic?`

Reply:
Perhaps "Unsupported intrinsic for DXIL lowering", but this seems informative enough.